src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -169,11 +169,15 @@ protected function convertValidationExceptionToResponse(ValidationException $e,
      */
     protected function prepareResponse($request, Exception $e)
     {
-        if ($this->isHttpException($e)) {
-            return $this->toIlluminateResponse($this->renderHttpException($e), $e);
-        } else {
+        if (! $this->isHttpException($e) && config('app.debug')) {
             return $this->toIlluminateResponse($this->convertExceptionToResponse($e), $e);
         }
+
+        if (! $this->isHttpException($e)) {
+            $e = new HttpException(500, $e->getMessage());
```

Comment:
If you want this in 5.4

Anyone who uses PHPStorm and auto-import suggest importing `HttpException` ([which might not exist](http://stackoverflow.com/questions/32191740/phpstorm-autocompletes-with-classes-that-do-not-actually-exist))

import `use Symfony\Component\HttpKernel\Exception\HttpException;` instead